backoff/retry.go

```diff
@@ -0,0 +1,135 @@
+package backoff
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"time"
+
+	retryablehttp "github.com/hashicorp/go-retryablehttp"
+)
+
+// HTTPClient is the interface of an http client that is compatible with
+// *http.Client.
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// RetryConfig configures the backoff-retry behaviour.
+type RetryConfig struct {
+	MinWait    time.Duration
+	MaxWait    time.Duration
+	MaxRetries int
+	CheckRetry retryablehttp.CheckRetry
+	Backoff    retryablehttp.Backoff
+}
+
+// DefaultRetryConfig is the default config for retrying http requests.
+var DefaultRetryConfig = RetryConfig{
+	MinWait:    1 * time.Second,
+	MaxWait:    30 * time.Second,
```

Comment:
I would like to see these default values in a config file rather than in-code.

Reply:
Still, even if you do not need to provide the configuration as a user, it would still be nice to have configuration separated from code and shipped with the library. 
This way, all the configuration could be seen at once and is not distributed around the codebase.

Reply:
added documentation for the RetryConfig in the README